src/CabinList.js

```diff
@@ -2,12 +2,21 @@ import React from 'react';
 import './App.css';
 import cabinData from './data/cabin';
 
-function Tweet(props) {
-  return (
-    <div className="tweet">
-      <h3>Cabin Names</h3>
-      {props.cabin}
-    </div>
-  );
+function CabinList() {
+  const click = (event) => {
+    console.log(event.target.innerHTML);
+  };
+  const cabinNameList1 = cabinData.map((data) => data.cabin);
+  const cabinNames = [...new Set(cabinNameList1)];
+  const cabin = cabinNames.map((data, index) => (
+    <li key={index}>
+      <button key={index} onClick={click}>
+        {data}{' '}
```

Comment:
lovely job.  But, we don't want to be relying on the text (innerHTML) of the button here, we want to make sure we're using exactly the same name text that's found in your data object.  

SO, try the following instead:

```
<button key={index} onClick={(e} => click(data)}>
  {data}
</button>
```

this is better because later on we'll be looking up your cabin data using this name, and, by doing this you use whatever you want for the button text.

You'll need to update your click function to match this.